scripts/deploy/src/install.js

```diff
@@ -0,0 +1,163 @@
+    }
+    return readFile(f);
+}
+
+function determineNamespace(values) {
+    const namespace = values.namespace || '';
+    if (!namespace) {
+        console.error("Namespace is not specified.");
+        process.exit(1);
+    }
+    return namespace;
+}
+
+function checkNamespaceExists(namespace) {
+    try {
+        execSync(`kubectl get namespace ${namespace}`, { stdio: 'inherit' });
+    } catch (err) {
+        console.log(`Namespace ${namespace} does not exist. Creating the namespace.`);
+        const namespaceManifest = `
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ${namespace}
+`;
+        const scriptDir = __dirname;
+        fs.writeFileSync(`${scriptDir}/helm/namespace.yaml`, namespaceManifest);
+        execSync(`kubectl apply -f ${scriptDir}/helm/namespace.yaml`, { stdio: 'inherit' });
+        fs.unlinkSync(`${scriptDir}/helm/namespace.yaml`);
+    }
+}
+
+async function obtainCertificateAndKey(values) {
+    console.log("Obtaining certificate...");
+    const certSource = values.cert_source || '';
+    if (certSource === 'specify-file-path') {
+        const crtFilePath = values.certificate_crt_file_path;
+        const keyFilePath = values.certificate_key_file_path;
+        if (!crtFilePath || !keyFilePath) {
+            throw new Error("certificate_crt_file_path and certificate_key_file_path must be set in values when cert_source is 'specify-file-path'");
+        }
+        fs.copyFileSync(crtFilePath, 'certificate.crt');
+        fs.copyFileSync(keyFilePath, 'private.key');
+    } else if (certSource === 'my-ip-co') {
+        const crtResponse = await axios.get('https://local-ip.medicmobile.org/fullchain');
+        const keyResponse = await axios.get('https://local-ip.medicmobile.org/key');
+        fs.writeFileSync('certificate.crt', crtResponse.data);
+        fs.writeFileSync('private.key', keyResponse.data);
+    } else if (certSource !== 'eks-medic') {
+        throw new UserRuntimeError("cert_source must be either 'specify-file-path', 'my-ip-co', or 'eks-medic'");
+    }
+}
+
+function createSecret(namespace, values) {
+    console.log("Checking if secret api-tls-secret already exists...");
+    try {
+        execSync(`kubectl get secret api-tls-secret -n ${namespace}`, { stdio: 'inherit' });
+    } catch (err) {
+        console.log("Secret does not exist. Creating Secret from certificate and key...");
+        obtainCertificateAndKey(values);
+        execSync(`kubectl -n ${namespace} create secret tls api-tls-secret --cert=certificate.crt --key=private.key`, { stdio: 'inherit' });
+        fs.unlinkSync('certificate.crt');
+        fs.unlinkSync('private.key');
+    }
+}
+
+async function get_image_tag(chtversion) {
+    const response = await axios.get(`https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:${chtversion}`);
+    const data = response.data;
+    const tag = data.tags.find(tag => tag.container_name === 'cht-api');
+    if (!tag) throw new Error('cht image tag not found');
```

Comment:
Thrown errors in asynchronous functions in javascript can behave a bit unexpected when coming from other languages, consider returning a rejected promise, error object, or exiting directly

Reply:
handled.